plugin/azure/azure.go

```diff
@@ -45,8 +46,11 @@ func New(ctx context.Context, dnsClient azure.RecordSetsClient, keys map[string]
 			if err != nil {
 				return nil, err
 			}
-			// Normalizing zoneName to make it fqdn
-			zoneNameFQDN := fmt.Sprintf("%s.", zoneName)
+			zoneNameFQDN := zoneName
+			// Normalizing zoneName to make it fqdn if required
+			if !strings.HasSuffix(zoneNameFQDN, ".") {
+				zoneNameFQDN = fmt.Sprintf("%s.", zoneNameFQDN)
+			}
 			if _, ok := zones[zoneNameFQDN]; !ok {
```

Comment:
you can just use dns.Fqdn(zonename) in this look up and delete all the lines above